app/Keycloak/TokenStrategy/ClientCredentials.php

```diff
@@ -0,0 +1,66 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Keycloak\TokenStrategy;
+
+use App\Json;
+use App\Keycloak\Dto\Config;
+use App\Keycloak\Dto\Realm;
+use App\Keycloak\Exception\KeyCloakApiFailed;
+use GuzzleHttp\ClientInterface;
+use GuzzleHttp\Exception\GuzzleException;
+
+/*
+ * LIMITATION: This class currently does not refresh the token automatically.
+ * For normal usage this should be ok, but if we ever implement long-running CLI processes this will need to be improved.
+ * */
+final class ClientCredentials implements TokenStrategy
+{
+    private ?string $accessToken = null;
+
+    public function __construct(
+        private readonly ClientInterface $client,
+        private readonly Config $config
+    ) {
+    }
+
+    public function fetchToken(Realm $realm): string
+    {
+        if ($this->accessToken === null) { //lazy loading
+            try {
+                if (!$this->config->isEnabled()) {
```

Comment:
I am not sure if this is the place to check the feature toggle if Keycloack is enabled or not. If you think about the final solution, it will be the creation of keys we want to enable. So looking at all other integrations this needs to be done inside the provider like here https://github.com/cultuurnet/publiq-platform/blob/main/app/Auth0/Auth0ServiceProvider.php#L92

Reply:
Removed the check